source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -82,10 +177,31 @@ void Cluster::onDnsHostAddOrUpdate(
   }
 
   ENVOY_LOG(debug, "adding new dfproxy cluster host '{}'", host);
+  Upstream::ClusterInfoConstSharedPtr cluster_info_to_use;
+  if (createCustomTlsForHost()) {
+    // Create an override cluster configuration that automatically provides both SNI as well as
+    // SAN verification for the resolved host if the cluster has been configured with TLS.
+    // TODO(mattklein123): The fact that we are copying the cluster config, etc. is not very clean.
+    //                     consider streamlining this in the future.
+    // TODO(mattklein123): If the host is an IP address should we be setting SNI? IP addresses in
```

Comment:
I recall that is not allowed, but @davidben or @PiotrSikora can confirm

Reply:
Yeah, IP addresses don't go in SNI, only the DNS names. This is a little goofy, but so it goes. (It means if you have an IP cert, you have to select it implicitly by way of connections without SNI. Although practically no one uses IP certs. :-/)

There were a few releases of Chrome where we accidentally put IP addresses in there, and it broke Apache httpd because they try to match SNI and Host header and got confused by IPv6 addresses. (All the colons and the existence of bracketed and unbracketed forms.)

Reply:
OK, thanks. If it's OK with everyone I'm going to fix this in my next PR in which I'm going to generally handle IP host headers correctly. So I will leave this as a TODO for my next PR.